discovery/consul/consul.go

```diff
@@ -50,7 +53,9 @@ func NewConsulConfig(config interface{}) (*Consul, error) {
 	if err != nil {
 		return nil, err
 	}
-	return &Consul{*client, false}, nil
+	upstreams := make(map[string][]*consul.ServiceEntry)
+	consul := &Consul{*client, false, sync.RWMutex{}, upstreams}
```

Comment:
Another nit... Shouldn't we unravel this a bit for the sake of clarity like we do within `configFromMap`. 

Something like...
```go
return &Consul{
	*client,
	wasRegistered: false,
	lock:          syncRWMutex{},
	upstreams:     upstreams,
}, nil
```
The reason I'm bringing it up is because I think the linter caught me doing this one time.